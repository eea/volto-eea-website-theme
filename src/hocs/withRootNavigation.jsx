import React from 'react';
import { connect, useDispatch } from 'react-redux';
import { getBaseUrl, hasApiExpander } from '@plone/volto/helpers';
import { getNavigation } from '@plone/volto/actions';
import config from '@plone/volto/registry';

export default function withRootNavigation(WrappedComponent) {
  return connect(
    (state) => {
      return {
        items: state.navigation.items,
        lang: state.intl.locale,
      };
    },
    { getNavigation },
  )((props) => {
    const dispatch = useDispatch();
    React.useEffect(() => {
      const { settings } = config;
      const currentLang = props.lang;
      const base_url = getBaseUrl(`/${currentLang}`);
      if (!hasApiExpander('navigation', base_url)) {
        dispatch(getNavigation(base_url, settings.navDepth));
      }
    }, [props.lang, dispatch]);

    return <WrappedComponent {...props} items={props.items} />;
  });
}
