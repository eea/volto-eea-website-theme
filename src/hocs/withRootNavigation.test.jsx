import React from 'react';
import withRootNavigation from './withRootNavigation';
import configureStore from 'redux-mock-store';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-intl-redux';
import { render } from '@testing-library/react';

const mockStore = configureStore();
let history = createMemoryHistory();

describe('withRootNavigation', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
    router: {
      location: {
        pathname: '/',
      },
    },
    navigation: {
      items: ['en'],
    },
  });
  const Block = (props) => {
    return (
      <>
        <div>My Block</div>
      </>
    );
  };
  it('injects navigation items into wrapped component', () => {
    const ExtendedComponent = withRootNavigation(Block);
    const items = [
      { title: 'page1', url: '/page1' },
      { title: 'page2', url: '/page2' },
    ];
    const { container } = render(
      <Provider store={store}>
        <Router history={history}>
          <ExtendedComponent items={items} />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
