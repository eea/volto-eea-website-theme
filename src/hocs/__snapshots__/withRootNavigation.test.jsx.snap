// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withRootNavigation injects navigation items into wrapped component 1`] = `
<div>
  <div>
    My Block
  </div>
</div>
`;
